app/server/models/user.py

```diff
@@ -499,9 +502,16 @@ def is_resetting(self):
         is_resetting = len(self.password_reset_tokens) > 0
         return is_resetting
 
-    # TODO(ussd): change to a field once we figure out what's the deal with resetting
     def pin_failed_attempts(self):
-        return 0
+        return self.failed_pin_attempts
+
+    @property
+    def pin(self):
+        return decrypt_string(self._encrypted_pin)
```

Comment:
hmm per the discussion here: https://trello.com/c/ei0HPPNB/32-implement-change-users-pin
we should be storing it salted (see `hash_password`). we should not be able to decrypt someone's pin

Reply:
So considering that hash_password returns no value and basically just updates the db's `password_hash` field. I think it would be more prudent to refactor the following into their own methods to make them reusable:
`bcrypt.hashpw(password.encode(), bcrypt.gensalt()).decode()`
`bcrypt.checkpw(password.encode(), self.password_hash.encode())`
right ?

Reply:
I really think we should consider just storing pins and PWs in the same field for now - it could keep the concept of "resetting" much simpler - means we don't need to think about whether they're resetting a pin or their password.

The argument against the above is that pins and PWs have different requirements (one is a set of digits, the other is utf-8), so perhaps need to have different treatment anyway.

Reply:
oh right. i forgot about this: @PhilipWafula if a user is resetting their pin, we also need to create a reset token and `save_password_reset_token`, and when they successfully set the pin, `delete_password_reset_tokens`. this is because in the `KenyaUssdProcessor`, we check if the user is currently resetting via `is_resetting` to send them through the right flow

@enjeyw i do agree that two edge cases could arise from this intertwined reset flow. if user is trying to reset their password, if mid-way through they use ussd, it'll send them down the reset pin flow. similarly, if they do intend to reset their pin mid-way through password reset, it'll void the token from their password reset. these seem veerrrry unlikely though, and i'd argue that if we do combine pin and password, if we ever want to separate them again it'll be pretty difficult if not impossible. and i can see a user having both a pin and a password for TFA

Reply:
or maybe it just makes sense to split up pin reset from password reset. pin reset being just a bool `is_resetting_pin`

Reply:
@PhilipWafula see this comment